lib/src/util.dart

```diff
@@ -83,9 +84,17 @@ final _oneOrMoreWhitespacePattern = RegExp('[ \n\r\t]+');
 
 /// "Normalizes" a link label, according to the [CommonMark spec].
 ///
-/// [CommonMark spec] https://spec.commonmark.org/0.29/#link-label
-String normalizeLinkLabel(String label) =>
-    label.trim().replaceAll(_oneOrMoreWhitespacePattern, ' ').toLowerCase();
+/// [CommonMark spec] https://spec.commonmark.org/0.30/#link-label
+String normalizeLinkLabel(String label) {
+  var text = label.trim().replaceAll(_oneOrMoreWhitespacePattern, ' ');
+  for (var i = 0; i < text.length; i++) {
+    final mapped = caseFoldingMap[text[i]];
+    if (mapped != null) {
+      text = text.replaceRange(i, i + 1, mapped);
```

Comment:
It's unfortunate that this is a little verbose, and `replaceRange` sounds like it can't be too performant, haha. Though maybe the backends have optimizations for ranges of size 1.

In any case, all of this code is only run for link labels, so, 🎉 , it's great.